bn-apps/cordapp-updates-distribution/corda-updates-transport/src/main/kotlin/net/corda/cordaupdates/transport/flows/PeekResourceFlow.kt

```diff
@@ -0,0 +1,63 @@
+package net.corda.cordaupdates.transport.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.cordaupdates.transport.flows.Utils.toCordaException
+import net.corda.cordaupdates.transport.flows.Utils.transporterFactory
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.flows.StartableByService
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.unwrap
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils
+import org.eclipse.aether.repository.RemoteRepository
+import org.eclipse.aether.spi.connector.transport.PeekTask
+import java.lang.Exception
+import java.net.URI
+
+/**
+ * Peeks resource in a remote repository. Used internally by Maven Resolver.
+ *
+ * @param resourceLocation location of the resource provided by Maven Resolver
+ * @param repoHosterName x500Name of the repository hoster
+ */
+@InitiatingFlow
+@StartableByRPC
+@StartableByService
+class PeekResourceFlow(private val resourceLocation : String, private val repoHosterName : String) : FlowLogic<Unit>() {
+    @Suspendable
+    override fun call() {
+        val repoHosterParty = serviceHub.identityService.wellKnownPartyFromX500Name(CordaX500Name.parse(repoHosterName))!!
+        val repoHosterSession = initiateFlow(repoHosterParty)
+        repoHosterSession.sendAndReceive<Boolean>(resourceLocation).unwrap { it }
+    }
+}
+
+/**
+ * This flows should exist at the repository hoster's node. The flow peeks an artifact in a configured file- or http(s)- based repository.
+ *
+ * The flow supports [SessionFilter]s to restrict unauthorised traffic.
+ */
+@InitiatedBy(PeekResourceFlow::class)
+class PeekResourceFlowResponder(session : FlowSession) : AbstractRepositoryHosterResponder<Unit>(session) {
+    @Suspendable
+    override fun doCall() {
+        val location = session.receive<String>().unwrap { it }
```

Comment:
Ditto ... can you refactor this "receive a URI and check it's valid" out into a shared function or superclass?

Also I'm not sure how this works. The remote end needs to know the `file://` URI of the repo on my own disk here? What if I want to move it? Shouldn't "location" be a GAV coordinate here, and the actual repo location is entirely the internal responsibility of the responding node?

Reply:
That's a bit weird thing about maven resolver. Transporters receive coordinates in the form of `net/corda/finance/1.5/finance-1.5.pom` - a relative coordinate of a file in a remote repository. This is why I'm a bit confused about how to validate it. One of the options would be convert it back to GAV I guess, which is a bit ugly.